### PR TITLE
Add hooks to csv driver

### PR DESCRIPTION
**Pull request by winfried-ripken**
# Description

Add hooks to csv driver to include simple preprocessing steps for quick loading of small csv datasets

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [ ] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

File: squirrel/driver/csv_driver.py [hunk 1]
```diff
@@ -14,14 +14,16 @@ class CsvDriver(FileDriver, DataFrameDriver):
 
     name = "csv"
 
-    def __init__(self, path: str, **kwargs) -> None:
+    def __init__(self, path: str, df_hooks: Iterable[Callable] | None = None, **kwargs) -> None:
         """Initializes CsvDriver.
 
         Args:
             path (str): Path to a .csv file.
+            df_hooks (Iterable[Callable], optional): preprocessing hooks to execute on the dataframe
```
Comment (AlirezaSohofi) [inline squirrel/driver/csv_driver.py:22]: Please mention that the first hook should accepts a single argument of type `dask.dataframe.DataFrame`. If that's not desirable, e.g. we want the hook to operate on pd.DataFrame, this need to be implemented too.

**Review by AlirezaSohofi — CHANGES_REQUESTED** [review]
LG. Just a note on the docstring.

**Comment by AlirezaSohofi** [comment]
@winfried-ripken  note this [comment](https://github.com/merantix-momentum/squirrel-datasets-core/pull/56#discussion_r982133207) from @jotterbach. Would make sense to give options to users to choose dask or pandas.

**Comment by winfried-ripken** [comment]
> @winfried-ripken note this [comment](https://github.com/merantix-momentum/squirrel-datasets-core/pull/56#discussion_r982133207) from @jotterbach. Would make sense to give options to users to choose dask or pandas.

Thanks for pointing this out!
I have added a flag to control this and changed the docstring where it was necessary :)

**Review by AlirezaSohofi — APPROVED** [review]
:+1:

**Pull request merged**